docs/KUBERNETES.md

```diff
@@ -187,6 +187,27 @@ Alternatively, update the DaemonSet configuration directly by using the followin
 kubectl patch daemonset -n logdna-agent logdna-agent --type json -p '[{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsUser","value":5000},{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsGroup","value":5000}]'
 ```
 
+### Enabling persistent agent state
+
+To avoid skipping or duplicating logs when the agent pod restarts or is replaced the agent needs to store it's progress on the host node's filesystem. This is achieved using a hostPath volume. The host directory must be writable by the user or group specified in the securityContext.
+
+To achieve this the host directory must either already exist with the correct
+permissions or else Kubernetes will create the directory without write permissions for the agent user.
+
+In this case the permissions must be set before the agent starts. When running as non-root the agent pod does not have permissions to do this, so an initcontainer may be used.
+
+Below is an example manifest section which can be added to the pod specification alongside the containers array. It assumes the agent user/group are both 5000 and the volume is mounted at /var/lib/logdna
+
```

Comment:
can we say "...an example manifest section of the -file-name-.yaml which can be added..." just to again be uber-clear about **where** this section lives? If you think everyone knows, skip it, but... ;-)

Also can we add a few more words to explain what 5000 means.. is this a permission level? Like the top level or...? I tried googling it and found nothing.

Reply:
I think the mention of "containers array"  is sufficient. 5000:5000 - just an example for setting ownership - user and group.